src/main/java/cms/gov/madie/measure/utils/JsonUtil.java

```diff
@@ -632,4 +644,53 @@ public static String getTestCaseJson(String json) throws JsonProcessingException
     }
     return null;
   }
+
+  // going through JsonNode, converts any datetime string into UTC datetime
+  public static JsonNode replaceNestedDateTimeStringValue(JsonNode node) {
+    if (node.isObject()) {
+      ObjectNode objectNode = (ObjectNode) node;
+      objectNode
+          .fields()
+          .forEachRemaining(
+              entry -> {
+                String fieldName = entry.getKey();
+                JsonNode childNode = entry.getValue();
+                String currentValue = childNode.asText();
+                String newValue = getNewValue(currentValue);
+                if (childNode.isTextual()) {
+                  objectNode.put(fieldName, newValue);
+                } else {
+                  replaceNestedDateTimeStringValue(childNode);
+                }
+              });
+    } else if (node.isArray()) {
+      node.forEach(childNode -> replaceNestedDateTimeStringValue(childNode));
+    }
+    return node;
```

Comment:
Do we need to return this node, Looks like recurssion calls are not capturing the returned value

Reply:
@RohitKandimalla This is removed.